gbpservice/neutron/nsf/configurator/agents/firewall.py

```diff
@@ -0,0 +1,323 @@
+# One Convergence, Inc. CONFIDENTIAL
+# Copyright (c) 2012-2016, One Convergence, Inc., USA
+# All Rights Reserved.
+#
+# All information contained herein is, and remains the property of
+# One Convergence, Inc. and its suppliers, if any. The intellectual and
+# technical concepts contained herein are proprietary to One Convergence,
+# Inc. and its suppliers.
+#
+# Dissemination of this information or reproduction of this material is
+# strictly forbidden unless prior written permission is obtained from
+# One Convergence, Inc., USA
+
+import os
+
+from oslo_config import cfg
+from oslo_log import log as logging
+import oslo_messaging as messaging
+from requests import ConnectionError
+
+from gbpservice.neutron.nsf.configurator.agents import agent_base
+from gbpservice.neutron.nsf.configurator.lib import fw_constants as const
+from gbpservice.neutron.nsf.configurator.lib import utils as load_driver
+from gbpservice.neutron.nsf.core import main
+
+
+LOG = logging.getLogger(__name__)
+
+rest_timeout = [
+    cfg.IntOpt(
+        'rest_timeout',
+        default=30,
+        help=_("rest api timeout"))]
+cfg.CONF.register_opts(rest_timeout)
+
+OPTS = [
+    cfg.IntOpt('oc_periodic_interval', default=10, help='Define periodic '
+                                                        'interval for tasks'),
+    cfg.IntOpt('oc_report_interval', default=10,
+               help='Reporting interval from agent to firewall plugin')
+]
+GROUP_OPTS = cfg.OptGroup(name='ocfwaas', title='OC FW OPTIONS')
+cfg.CONF.register_group(GROUP_OPTS)
+cfg.CONF.register_opts(OPTS, group=GROUP_OPTS)
+
+
+"""Implements APIs invoked by configurator for processing RPC messages.
+
+RPC client of configurator module receives RPC messages from REST server
+and invokes the API of this class. The instance of this class is registered
+with configurator module using register_service_agent API. Configurator module
+identifies the service agent object based on service type and invokes ones of
+the methods of this class to configure the device.
+
+"""
+
+
+class FwaasRpcSender(object):
+    """ APIs to FWaaS Plugin.
+    """
+
+    def __init__(self, sc, host, nqueue):
+        self.sc = sc
+        self.host = host
+        self.qu = nqueue
+
+    def set_firewall_status(self, context, firewall_id, status):
+        """Make an RPC to set the status of a firewall."""
+
+        msg = {'receiver': const.NEUTRON,
+               'resource': const.SERVICE_TYPE,
+               'method': 'set_firewall_status',
+               'data': {'context': context,
+                        'host': self.host,
+                        'firewall_id': firewall_id,
+                        'status': status}
```

Comment:
Fwaas response doesn't go to orchestrator right?

Regards,
Deepak

On Tue, Feb 23, 2016 at 10:41 PM, njagadish <notifications@github.com>
wrote:

> In gbpservice/neutron/nsf/configurator/agents/firewall.py
> <https://github.com/oneconvergence/group-based-policy/pull/71#discussion_r53813725>
> :
>
> > +
> > +    def __init__(self, sc, host, nqueue):
> > +        self.sc = sc
> > +        self.host = host
> > +        self.qu = nqueue
> > +
> > +    def set_firewall_status(self, context, firewall_id, status):
> > +        """Make an RPC to set the status of a firewall."""
> > +
> > +        msg = {'receiver': const.NEUTRON,
> > +               'resource': const.SERVICE_TYPE,
> > +               'method': 'set_firewall_status',
> > +               'data': {'context': context,
> > +                        'host': self.host,
> > +                        'firewall_id': firewall_id,
> > +                        'status': status}
>
> We need to add request_info dict to the notification. It is needed for
> orchestartor to map a request to response.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/oneconvergence/group-based-policy/pull/71/files#r53813725>
> .
>